### PR TITLE
Csv upload

### PR DESCRIPTION
**Pull request by JerameyATyler**
Added functionality to upload a csv on the numeric gradeable data entry page. Only users with instructor level privileges will be able to access this functionality.

### CODE REVIEW

File: TAGradingServer/account/ajax/account-numerictext-gradeable.php [hunk 1]
```diff
@@ -61,33 +76,41 @@
...
     }
 }
 
+function csvupload(){
+
+    $csv = $_POST['parsedCsv'];
+    global $db;
+    $params = array('{"'.implode('","', $csv).'"}', $_GET["g_id"], User::$user_id);
+
+    $db->query("SELECT * FROM csv_To_Numeric_Gradeable(?, ?, ?)", $params);
```
Comment (MasterOdin) [inline TAGradingServer/account/ajax/account-numerictext-gradeable.php]: Wouldn't this make more sense to do `SELECT csv_To_Numeric(?, ?, ?)`?

And what's the benefit of doing all of this stuff directly in the database as a function than using PHP & DB to handle this stuff? Using purely DB does make it a tad harder to easily update it if we decide to change some part of the code later.

File: TAGradingServer/account/account-numerictext-gradeable.php [hunk 2]
```diff
@@ -21,6 +21,52 @@
...
+                reader.readAsText(f);
+                reader.onload = function(evt) {
+                  parseCsv(reader.result);
+                  
+                }
+                reader.onerror = function(evt){
+                    console.error(\"nope\");
+                }
+            } 
+        }
```
Comment (MasterOdin) [inline TAGradingServer/account/account-numerictext-gradeable.php]: I'd rather just pass a $_FILES object to be handled by PHP and not JS, especially as we're not really doing anything with the parsed object other than saying "we could open this file!"

File: TAGradingServer/account/account-numerictext-gradeable.php [hunk 3]
```diff
@@ -21,6 +21,52 @@
...
+        
+        function parseCsv(csv){
+            url = \"{$BASE_URL}/account/ajax/account-numerictext-gradeable.php?course={$_GET['course']}&semester={$_GET['semester']}&g_id={$_GET['g_id']}\";
+            var lines = csv.split(/\\r\\n|\\n/);
+            console.log(lines);
+            console.log(url);
```
Comment (MasterOdin) [inline TAGradingServer/account/account-numerictext-gradeable.php]: remove these `console.log` lines

File: TAGradingServer/account/account-numerictext-gradeable.php [hunk 4]
```diff
@@ -21,6 +21,52 @@
...
+                    action:\"csv\"
+                },
+                success: function(data, text){
+                    location.reload();
+                },
+                error: function(request, status, error){
+                    window.alert(\"An error has occurred. Contact an administrator.\");
+                }
+            });
+        }";
```
Comment (MasterOdin) [inline TAGradingServer/account/account-numerictext-gradeable.php]: So I was talking about [Heredoc](http://php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc)

In that it allows you to do:
```
$a = <<<HTML
Hello "{$variable}", we're using double quotes and variables and no need for escaping things!
Woo!
HTML;
```

**Comment by bmcutler** [comment]
merging this now...   but please continue the conversation in issue #721....

**Pull request merged**